cmd/anti_affinity_group_create.go

```diff
@@ -42,26 +39,43 @@ func (c *antiAffinityGroupCreateCmd) cmdPreRun(cmd *cobra.Command, args []string
 }
 
 func (c *antiAffinityGroupCreateCmd) cmdRun(_ *cobra.Command, _ []string) error {
-	zone := account.CurrentAccount.DefaultZone
+	ctx := gContext
 
-	ctx := exoapi.WithEndpoint(gContext, exoapi.NewReqEndpoint(account.CurrentAccount.Environment, zone))
-
-	antiAffinityGroup := &egoscale.AntiAffinityGroup{
-		Description: utils.NonEmptyStringPtr(c.Description),
-		Name:        &c.Name,
+	antiAffinityGroupCreateRequest := v3.CreateAntiAffinityGroupRequest{
+		Description: c.Description,
+		Name:        c.Name,
 	}
 
-	var err error
-	decorateAsyncOperation(fmt.Sprintf("Creating Anti-Affinity Group %q...", c.Name), func() {
-		antiAffinityGroup, err = globalstate.EgoscaleClient.CreateAntiAffinityGroup(ctx, zone, antiAffinityGroup)
+	err := decorateAsyncOperations(fmt.Sprintf("Creating Anti-Affinity Group %q...", c.Name), func() error {
+		op, err := globalstate.EgoscaleV3Client.CreateAntiAffinityGroup(ctx, antiAffinityGroupCreateRequest)
+		if err != nil {
+			return fmt.Errorf("exoscale: error while creating Anti Affinity Group: %w", err)
+		}
+
+		_, err = globalstate.EgoscaleV3Client.Wait(ctx, op, v3.OperationStateSuccess)
+		if err != nil {
+			return fmt.Errorf("exoscale: error while waiting for Anti Affinity Group creation: %w", err)
+		}
+
+		return nil
 	})
 	if err != nil {
 		return err
 	}
 
+	antiAffinityGroupsResp, err := globalstate.EgoscaleV3Client.ListAntiAffinityGroups(ctx)
+	if err != nil {
+		return err
+	}
+
+	antiAffinityGroup, err := antiAffinityGroupsResp.FindAntiAffinityGroup(c.Name)
+	if err != nil {
+		return err
+	}
+
 	return (&antiAffinityGroupShowCmd{
```

Comment:
same than other PR for `antiAffinityGroupShowCmd` type could be adapted here :)

Reply:
Here unfortunately, the `AntiAffinityGroup` needs to be a `string` in `antiAffinityGroupShowCmd`